File: AoC2019/Python3/day3.py
import numpy as np
#puzzle setup
with open('./day3in.txt','r') as pInput:
    pInput = pInput.read().splitlines()

#do stuff
wireA = pInput[0].split(',')
wireB = pInput[1].split(',')
wacoords = [(0,0)]
wbcoords = [(0,0)]
dirs = 'UDLR'
dircoords = [(0,1),(0,-1),(-1,0),(1,0)]

def wireStep(go,wire):
    for y in range(0,go):
        zips = zip(wire[-1],dircoords[dirs.index(x[:1])])
        wire.append(tuple([sum(zipped) for zipped in zips]))

for x in wireA:
    wireStep(int(x[1:]),wacoords)

for x in wireB:
    wireStep(int(x[1:]),wbcoords)

intersect = list(set(wacoords).intersection(set(wbcoords)))
intersect.remove((0,0))
print(np.sum(np.abs(min(intersect,key=lambda x: sum(map(abs,x))))))

File: AoC2015/JavaScript/day3.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('./day3in.txt', 'utf8');
input = input.split('');

//do stuff
ans = [[0,0,1]];
pos = 0;

for (let i = 0; i < input.length; i++)
{
    let temp = [0];
    temp.unshift(ans[pos][0],ans[pos][1]);
    switch(input[i])
    {
        case "^":
            temp[1]++;
            break;
        case "v":
            temp[1]--;
            break;
        case "<":
            temp[0]--;
            break;
        case ">":
            temp[0]++;
            break;
    }
    beenThere(temp);
}
console.log(ans.length)

function beenThere(arr)
{
    for (i in ans)
    {
        if (ans[i][0]==arr[0]&&ans[i][1]==arr[1])
        {
            ans[i][2]++;
            pos = i;
            return;
        }
    }
    arr[2] = 1;
    ans.push(arr);
    pos = ans.length - 1;
    return;
}